joomla/joomla-cms
# Bootstrap popup with mootols compatibility

### Joomla 3.5 should had bootstrap modal for form field media but didn't. This fixes that
#### Summary of Changes

Introduce an option that will provide the needed backwards compatibility.
#### Testing Instructions

Apply patch
- Bootstrap mode
  Test creating a new article and that the intro/full text images as well as the tinymce image button still works.
- Mixed mode
  Edit administrator/components/com_content/views/article/tmpl/edit.php
  and paste the code bellow after line 84 (just before `?>`)

``` php
JHtml::_('behavior.modal');

$idTag = 'jform_name';
$remoteUrl = 'index.php?option=com_media&amp;view=images&amp;tmpl=component&amp;asset=com_users&amp;author=' . JFactory::getUser()->id . '&amp;fieldid=' . $idTag;
$buttonId = $idTag . '_btn';

JHtml::_('bootstrap.modal');
$test_control[] = '<a id="' . $buttonId . '" href="#' . $idTag . '_modal" role="button" class="btn" data-toggle="modal" title="' . JText::_('JSELECT') . '">SELECT CUSTOM BS</a>';
$test_control[] = JHtmlBootstrap::renderModal(
    $idTag . '_modal',
    array(
        'url' => $remoteUrl,
        'title' => JText::_('JSELECT'),
        'height' => '600px', 'width' => '500px')
);

$test_control[] = '<input type="text" title="cscsc" id="jform_name" >';

$test_control[] = '<a class="modal btn" title="XXXXX" href="index.php?option=com_media&amp;view=images&amp;tmpl=component&amp;asset=image&amp;author='
    . JFactory::getUser()->id . '&amp;fieldid=jform_name&amp;folder="'
    . ' rel="{handler: \'iframe\', size: {x: 800, y: 500}}">SELECT MOOTOOLS</a>';

echo implode('', $test_control);

//JFactory::getDocument()->addScriptDeclaration("
//    window.jInsertFieldValue = function(value, id) {jQuery('#' + id).val(value);};
//    ");
```

Test again the previous scenario as well as the new buttons that will appear above the tabs. The field should be field with the image selected in the pop up
- Compatibility Mode (AKA Mootools)
  Keep the changes in edit.php
  Rename administrator/templates/isis/html/layouts/joomla/form/field/media.php to xxxmedia.php
  Repeat the previous testing steps
